### PR TITLE
format: Reformat Robot Framework files with robotframework-tidy

### PR DESCRIPTION
**Pull request by Bravo555**
## Proposed changes

Reformat Robot Framework files with a standard Robot Framework formatter, [robotframework-tidy](https://robotidy.readthedocs.io/en/stable/index.html).

The motivation is for our Robot Framework test suite to be more readable and consistent, by having the formatting be consistent and adhere to [Robot Framework Style Guide](https://docs.robotframework.org/docs/style_guide).

Most of the team writes Robot Framework tests in tandem with features they're working on, so they most often just expand existing test suites and replicate whatever style is already present there, but sometimes new suites are still created, also with different styles. As all of these suites expand, the styles can gradually drift apart.

Eventually perhaps everybody would set up their development environment to format these files automatically and we could set up formatting checks as we have for Rust, but that would require some effort for everybody to set things up, for now we can do this one-time reformatting.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue

https://github.com/thin-edge/thin-edge.io/issues/2521

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

We could also use [Robocop](https://robocop.readthedocs.io/en/stable/), but even after reformatting with Robotidy, it finds a lot of warnings that would have to be resolved manually:

```
$ robocop --report rules_by_error_type

...

Found 1942 issues: 1754 WARNINGs, 188 INFOs.
```

It would perhaps also require a less trivial restructuring of the test suite, which would be a lot of manual work, and it also finds a lot of `Line is too long (203/120) (line-too-long)` warnings, which should be handled by robotidy but apparently weren't. But the option is there.

### CODE REVIEW

**Comment by reubenmiller** [comment]
@Bravo555 great initiative.

I agree that we can phase in this formatting, so a once off effort is ok, and then we can gain experience (and it is ok), then we can add it to the workflow checks (as you suggested).

**Comment by reubenmiller** [comment]
@Bravo555 I've added robotframework-robocop to the dev dependencies and added the following pyinvoke tasks (to run both linting and formatting manually).

```sh
invoke lint-tests
invoke format-tests
```

Since we have a lot of linting errors at the moment, I've disabled the cosmetic ones (e.g. line size, missing doc strings etc.) and then fixed some of the more pressing issues. This will give us a good starting point, and in time, we can then go through the linting rules one by one and enable them again and address the rules etc.

**Review by reubenmiller — APPROVED** [review]
Good improvement and brings in some formatting consistency. We can slowly fix the remaining linting rules (which are currently disabled in the pyproject.toml as there were too many warnings to address immediately).

@Bravo555 Though review some of my commits which add the invoke tasks and robotframework-robocop dependency etc.

**Comment by albinsuresh** [comment]
Thank you @Bravo555 for the initiative. Once merged, you may close this ticket as well: https://github.com/thin-edge/thin-edge.io/issues/2521

**Comment by reubenmiller** [comment]
> Thank you @Bravo555 for the initiative. Once merged, you may close this ticket as well: #2521

Thanks for the reminder, I just added the link to the ticket description as well

File: tests/RobotFramework/tests/tedge_agent/device_profile/device_profile_operation.robot [hunk 1]
```diff
@@ -1,48 +1,52 @@
...
+    ...    "url":""
+    ...    }
+    ...    ],
+    ...    "configuration":[
+    ...    {
+    ...    "name":"tedge-configuration-plugin",
+    ...    "type":"tedge-configuration-plugin",
+    ...    "url":"${config_url}"
+    ...    }
+    ...    ]
```
Comment (albinsuresh) [inline tests/RobotFramework/tests/tedge_agent/device_profile/device_profile_operation.robot:49]: Hmm... Not great, but I understand why.

**Review by albinsuresh — APPROVED** [review]
LGTM. It'd be great if the linter and formatter commands are integrated into the `just` recipes as well.

File: tests/RobotFramework/tests/tedge_agent/device_profile/device_profile_operation.robot (hunk 1 above)
Comment (reubenmiller) [reply tests/RobotFramework/tests/tedge_agent/device_profile/device_profile_operation.robot:49]: Yeah I think we can add a todo to investigate a better way to format it so it is more readable in a future PR

File: tests/RobotFramework/tasks.py [hunk 2]
```diff
@@ -125,12 +125,27 @@ def lint(c):
     """Run linter"""
     c.run(f"{sys.executable} -m pylint libraries")
 
+@task
+def lint_tests(c, threshold="W"):
+    """Run RobotFramework Test Linter
+
+    Example:
+        $ invoke lint-tests --threshold I
```
[the remark this answers is not in the document]
Comment (Bravo555) [reply tests/RobotFramework/tasks.py:133]: c04e25e501

File: tests/RobotFramework/tasks.py [hunk 3]
```diff
@@ -125,12 +125,27 @@ def lint(c):
...
 @task(name="format")
 def formatcode(c):
     """Format python code"""
     c.run(f"{sys.executable} -m black libraries")
 
+@task
+def format_tests(c):
+    """Format RobotFramework tests
+    """
+    c.run(f"{sys.executable} -m robotidy tests")
```
[the remark this answers is not in the document]
Comment (Bravo555) [reply tests/RobotFramework/tasks.py:147]: c04e25e501

**Review by didier-wenzek — APPROVED** [review]
Approved. Thank you for the initiative.

**Pull request merged**